scipy/integrate/_ivp/ivp.py

```diff
@@ -18,12 +19,53 @@
 
 
 MESSAGES = {0: "The solver successfully reached the end of the integration interval.",
-            1: "A termination event occurred."}
+            1: "A termination event occurred.",
+            2: "Solver stopped via callback."}
 
 
 class OdeResult(OptimizeResult):
     pass
 
+class IntermediateOdeResult(_RichResult):
```

Comment:
`OdeResult` above does not have documentation (only in the `Returns` section of `solve_ivp`). It would be useful to have documentation for `OdeResult` for comparison purposes and linking.

It is also weird to me that it inherits from `OptimizeResult` rather than from `_RichResult` directly.

Reply:
It's probably a matter of history. `OptimizeResult` existed first; `_RichResult` was created much later.